### PR TITLE
Reject reserved words

### LINKED ISSUE #429: VM build fails when using C-reserved words as temp variable names
While working on bytecode operations (see `initializeBytecodeTableForSistaV1`), I used `register` as a temp variable name in one of the bytecode handlers.

At the moment of building the VM, I got a parse error, as `register` is a reserved word in C.

I think we should detect those cases, and add a prefix/suffix to those variable names so that they do not collide with C reserved words.

### PR DESCRIPTION
**Pull request by ivojawer**
fixes #429 

- Checks for keywords in
  - Arguments
  - Selector
  - Variable (instance, parameters, temporary) names
- Changed parameters named 'register' to 'reg'
- Changed `CogSimStackEntry` Selector + accessors named `register` to `registerr`

### CODE REVIEW

**Review by guillep — APPROVED** [review]
Thanks!!

**Comment by guillep** [comment]
4 tests are broken, related to this change

imagen

`Message not understood: CogSimStackEntry >> #register`

File: smalltalksrc/Slang/TVariableNode.class.st [hunk 1]
```diff
@@ -127,7 +127,11 @@ TVariableNode >> printOn: aStream level: level [
 { #category : #accessing }
 TVariableNode >> setName: aString [
 
-	name := aString.
+	(CCodeGenerator reservedWords includes:
+		 aString) ifTrue: [
+		TranslationError signal: aString , ' is a reserved word' ].
```
Comment (PalumboN) [inline smalltalksrc/Slang/TVariableNode.class.st:132]: For me, this is _polémico_. Do we want this validation in the setter? I was thinking to have this validation before the next transformation. But I don't know how it is used...

Also, we can have a `validateReservedWords` method as in `TMethod`.

File: smalltalksrc/VMMaker/CogSimStackEntry.class.st [hunk 2]
```diff
@@ -76,11 +76,11 @@ CogSimStackEntry class >> instVarNamesAndTypesForTranslationDo: aBinaryBlock [
 	self filteredInstVarNames do:
 		[:ivn|
 		aBinaryBlock
-			value: (ivn = 'register' ifTrue: ['registerr'] ifFalse: [ivn]) "avoid reservedWord conflict"
```
Comment (PalumboN) [inline smalltalksrc/VMMaker/CogSimStackEntry.class.st:79]: image

**Review by PalumboN — APPROVED** [review]
Niceeeeeeee 🤩 

Big renames

**Comment by PalumboN** [comment]
Tests passing, ready to merge

**Comment by guillep** [comment]
Thanks @ivojawer !!

**Pull request merged**